### PR TITLE
[firebase-recaptcha] Add appVerificationDisabledForTesting prop to enable test mode

### LINKED ISSUE #10716: expo-firebase-recaptcha should include appVerificationDisabledForTesting
Hello Guys!

Been using expo-firebase-recaptcha and works amazing with phone auth. 

But while in development is a pain in the ass

firebase already has an option called appVerificationDisabledForTesting that will disable recaptcha for whitelisted numbers

https://rnfirebase.io/reference/auth/authsettings#appVerificationDisabledForTesting

I guess this prop should be included, right now im using patch-package as a workaround to add 

`firebase.auth().settings.appVerificationDisabledForTesting = true;`

Greetings!

### PR DESCRIPTION
**Pull request by IjzerenHein**
# Why

This PR adds the prop `appVerificationDisabledForTesting` which causes reCAPTCHA to render a mock UI which makes testing a lot easier to do.

Fixes #10716

ezgif com-gif-maker (1)

# How

- Add `appVerificationDisabledForTesting` prop
- Update docs
- Updated NCL
- Update changelog

# Test Plan

- Start NCL `cd apps/native-component-list && expo start`
- Open `API -> Firebase reCAPTCHA`
- Click on 2nd list-item row to enable `appVerificationDisabledForTesting `
- Click on 1st list-item row to request a reCAPTCHA token
- Token is obtained without requiring the user to complete the challenge

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]

File: docs/pages/versions/unversioned/sdk/firebase-recaptcha.md [hunk 1]
```diff
@@ -256,6 +256,7 @@ Modal screen that is automatically shown and displays a reCAPTCHA widget. The re
 
 - **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
 - **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`.
+- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this is disabled, a mock reCAPTCHA is rendered instead. This is useful for manual testing during development or for automated integration tests.
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/firebase-recaptcha.md:259]:
```suggestion
- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this prop is `true`, a mock reCAPTCHA is rendered. This is useful for manual testing during development or for automated integration tests.
```

maybe good to link to the firebase docs for this prop as well for more info

File: docs/pages/versions/unversioned/sdk/firebase-recaptcha.md [hunk 2]
```diff
@@ -267,6 +268,7 @@ The reCAPTCHA v3 widget displayed inside a web-view.
 
 - **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
 - **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`.
+- - **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this is disabled, a mock reCAPTCHA is rendered instead. This is useful for manual testing during development or for automated integration tests.
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/firebase-recaptcha.md:271]:
```suggestion
- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this prop is `true`, a mock reCAPTCHA is rendered. This is useful for manual testing during development or for automated integration tests.
```

same as above, also this one has an extra `-` in front :O

File: docs/pages/versions/v39.0.0/sdk/firebase-recaptcha.md [hunk 3]
```diff
@@ -256,6 +256,7 @@ Modal screen that is automatically shown and displays a reCAPTCHA widget. The re
 
 - **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
 - **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`.
+- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this is disabled, a mock reCAPTCHA is rendered instead. This is useful for manual testing during development or for automated integration tests.
```
Comment (brentvatne) [inline docs/pages/versions/v39.0.0/sdk/firebase-recaptcha.md:259]:
```suggestion
- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this prop is `true`, a mock reCAPTCHA is rendered. This is useful for manual testing during development or for automated integration tests.
```

File: docs/pages/versions/v39.0.0/sdk/firebase-recaptcha.md [hunk 4]
```diff
@@ -267,6 +268,7 @@ The reCAPTCHA v3 widget displayed inside a web-view.
 
 - **firebaseConfig (IFirebaseOptions)** -- Firebase web configuration.
 - **firebaseVersion (string)** -- Optional version of the Firebase JavaScript SDK to load in the web-view. You can use this to load a custom or newer version. For example `version="6.8.0"`.
+- - **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this is disabled, a mock reCAPTCHA is rendered instead. This is useful for manual testing during development or for automated integration tests.
```
Comment (brentvatne) [inline docs/pages/versions/v39.0.0/sdk/firebase-recaptcha.md:271]:
```suggestion
- **appVerificationDisabledForTesting (boolean)** -- When set, this property disables app verification for the purpose of testing phone authentication. When this prop is `true`, a mock reCAPTCHA is rendered. This is useful for manual testing during development or for automated integration tests.
```

File: docs/pages/versions/unversioned/sdk/firebase-recaptcha.md (hunk 1 above)
Comment (IjzerenHein) [reply docs/pages/versions/unversioned/sdk/firebase-recaptcha.md]: Thanks, I've added a link to firebase as well 👍

**Pull request merged**